lib/oli/delivery/attempts/core.ex

```diff
@@ -224,37 +224,31 @@ defmodule Oli.Delivery.Attempts.Core do
 
   `[%ResourceAccess{}, ...]`
   """
-  def get_graded_resource_access_for_context(section_slug) do
+  def get_graded_resource_access_for_context(section_id) do
```

Comment:
Wdyt about writing a single query and reusing it in both methods since they are exactly the same? And in the below one, we can add the extra where condition dynamically

Reply:
Oh yeah good call.  @mcass19 take another look, I reworked this into a base, composable query that adds in the user id condition only for the 2-arity instance of the function.

Reply:
Looks great!